webapp-django/crashstats/authentication/migration.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+User = get_user_model()
+
+
+def migrate_users(combos, dry_run=False):
+    # Because the list of comboes is possibly very large, we don't want
+    # to query the ORM for each and every one since most of the emails
+    # we expect to search for are not going to be found.
+    size = 100
+    groups = [
+        combos[i:i + size] for i in range(0, len(combos), size)
+    ]
```

Comment:
I always found it puzzling that this isn't in the Python itertools module. Because of that, I keep writing it:

https://github.com/mozilla/fjord/blob/master/fjord/search/index.py#L249

Erik wrote one, too:

https://github.com/erikrose/more-itertools/blob/master/more_itertools/more.py#L15

Yours is fine, too. It just always puzzled me.